src/hotspot/share/gc/serial/markSweep.cpp

```diff
@@ -214,4 +216,5 @@ void MarkSweep::KeepAliveClosure::do_oop(narrowOop* p) { MarkSweep::KeepAliveClo
 void MarkSweep::initialize() {
   MarkSweep::_gc_timer = new (ResourceObj::C_HEAP, mtGC) STWGCTimer();
   MarkSweep::_gc_tracer = new (ResourceObj::C_HEAP, mtGC) SerialOldTracer();
+  MarkSweep::_string_dedup_requests = new StringDedup::Requests();
```

Comment:
It's not clear to me why `_string_dedup_requests` is a pointer. Is it possible to define `static StringDedup::Requests _string_dedup_requests;` directly? That way, `Requests` doesn't need to use the inheritance.

Reply:
It's a non-local variable with a non-trivial destructor.  See
https://mail.openjdk.java.net/pipermail/hotspot-gc-dev/2021-August/036412.html